pkg/capacityscheduling/capacity_scheduling.go

```diff
@@ -0,0 +1,882 @@
+			continue
+		}
+		potentialNodes = append(potentialNodes, node)
+	}
+	return potentialNodes
+}
+
+func selectVictimsOnNode(
+	ctx context.Context,
+	ph framework.PreemptHandle,
+	state *framework.CycleState,
+	pod *v1.Pod,
+	nodeInfo *framework.NodeInfo,
+	pdbs []*policy.PodDisruptionBudget,
+	snapshotState *SnapshotState,
+	preFilterState *PreFilterState,
+) ([]*v1.Pod, int, bool) {
+	var potentialVictims []*v1.Pod
+	var moreThanMinWithPreemptor bool
+
+	removePod := func(rp *v1.Pod) error {
+		if err := nodeInfo.RemovePod(rp); err != nil {
+			return err
+		}
+		status := ph.RunPreFilterExtensionRemovePod(ctx, state, pod, rp, nodeInfo)
+		if !status.IsSuccess() {
+			return status.AsError()
+		}
+		return nil
+	}
+	addPod := func(ap *v1.Pod) error {
+		nodeInfo.AddPod(ap)
+		status := ph.RunPreFilterExtensionAddPod(ctx, state, pod, ap, nodeInfo)
+		if !status.IsSuccess() {
+			return status.AsError()
+		}
+		return nil
+	}
+
+	podPriority := podutil.GetPodPriority(pod)
+	preemptorElasticQuotaInfo, preemptorWithElasticQuota := snapshotState.elasticQuotas[pod.Namespace]
+
+	// Check if there is elastic quota in the preemptor's namespace.
+	if preemptorWithElasticQuota {
+		moreThanMinWithPreemptor = preemptorElasticQuotaInfo.moreThanMinWithPod(preFilterState.Resource)
+	}
+
+	for _, p := range nodeInfo.Pods {
+		pElasticQuotaInfo, pWithElasticQuota := snapshotState.elasticQuotas[p.Pod.Namespace]
+		// Preemption only happens between in the same type of namespace: with or without.
+		if preemptorWithElasticQuota == pWithElasticQuota {
+			// If there is elastic quota in the preemptor's namespace.
+			if preemptorWithElasticQuota {
+				if moreThanMinWithPreemptor {
+					// If Preemptor.Request + Quota.allocated > Quota.min:
+					// It means that its guaranteed isn't borrowed by other
+					// quotas. So that we will select the pods belongs to the
+					// same quota(namespace) with the lower priority than the
+					// preemptor’s priority as potential victims in a node.
+					if p.Pod.Namespace == pod.Namespace && podutil.GetPodPriority(p.Pod) < podPriority {
+						potentialVictims = append(potentialVictims, p.Pod)
+						if err := removePod(p.Pod); err != nil {
+							return nil, 0, false
+						}
+					}
+				} else {
+					// If Preemptor.Request + Quota.allocated <= Quota.min: It
+					// means that its min or guaranteed resource is used or
+					// `borrowed` by other Quota. Potential victims in a node
+					// will be chosen from Quotas that allocates more resources
+					// than its min, i.e., borrowing resources from other
+					// Quotas.
+					if pElasticQuotaInfo.overUsed() &&
+						!pElasticQuotaInfo.lessThanMinWithoutPod(computePodResourceRequest(pod).Resource) {
+						potentialVictims = append(potentialVictims, p.Pod)
+						if err := removePod(p.Pod); err != nil {
+							return nil, 0, false
+						}
+					}
+				}
+			} else {
+				if podutil.GetPodPriority(p.Pod) < podPriority {
```

Comment:
Does this mean you'd mixin the defaultpreemption behavior? So that you'd like to disable defaultpreemption upon configuration?

Reply:
Yes. I disable defaultpreemption upon configruration.
```go
    postFilter:
      enabled:
      - name: CapacityScheduling
      disabled:
      - name: "*"
```